docs/asyncio_thread_safety.md

```diff
@@ -0,0 +1,163 @@
+---
+title: "Thread Safety with asyncio"
+---
+
+Developing with asyncio requires careful attention to thread safety, as nearly all asyncio objects are not thread-safe. If you are just getting started with asyncio, review Python's documentation on [Developing with asyncio](https://docs.python.org/3/library/asyncio-dev.html) for tips to avoid pitfalls. 
+
+Home Assistant has some conventions for handling async and non-async code in the same code base. The top highlights are:
+
+- Deciding how to run a function from a helper depends on whether it is decorated with `@callback`; for more details, see [Working with Async](asyncio_working_with_async).
```

Comment:
I would make this a tiny bit clearer to the point. Decorating a method with @callback indicates its safe to call from the eventloop (thread).

Reply:
Adjusted in d8bc1190cd6838c9e164538b85ed9b8a00b6e31b